src/taipy/gui_core/GuiCoreLib.py

```diff
@@ -212,7 +253,15 @@ def on_init(self, gui: Gui) -> t.Optional[t.Tuple[str, t.Any]]:
         return GuiCore.__CTX_VAR_NAME, GuiCoreContext(gui)
 
     def on_user_init(self, state: State):
-        state._add_attribute(GuiCoreContext._ERROR_VAR)
-        state._gui._bind_var_val(GuiCoreContext._ERROR_VAR, "")
-        state._add_attribute(GuiCoreContext._SCENARIO_SELECTOR_ID_VAR)
-        state._gui._bind_var_val(GuiCoreContext._SCENARIO_SELECTOR_ID_VAR, "")
+        for var in [
+            GuiCoreContext._SCENARIO_SELECTOR_ERROR_VAR,
+            GuiCoreContext._SCENARIO_SELECTOR_ID_VAR,
+            GuiCoreContext._SCENARIO_VIZ_ERROR_VAR,
+        ]:
+            state._add_attribute(var)
```

Comment:
Does it make sense to add_attribute() but not bind_var_val()?
In which case a state._add_attribute(gui, var) would certainly be simpler...

Reply:
I've updated _add_attribute on the current State PR (feature/#737-state.refresh) 
I'll modify this once that PR is merged